tools/cli/helpers/doc-parser/src/class-doc-parser.php

```diff
@@ -446,15 +458,19 @@ public function pretty_print_hook_name( Node\Arg $argument ): string {
 				$part = $value->{$property};
 				if ( $part instanceof Node\Scalar\String_ ) {
 					$result .= $part->value;
-				} elseif ( $part instanceof Node\Expr ) {
+				} elseif ( $part instanceof Node\Expr ) { // @phan-suppress-current-line PhanRedundantConditionInLoop
 					$result .= '{' . $this->printer->prettyPrint( array( $part ) ) . '}';
 				} else {
-					throw new \Exception( 'Unexpected concatenated string component of type ' . get_class( $part ) );
+					throw new \UnexpectedValueException( 'Unexpected concatenated string component of type ' . get_class( $part ) );
 				}
 			}
 			return $result;
+		} elseif ( $argument->value instanceof Node\Expr\Variable ) {
+
+			// We don't care about variable names, we can't really document them.
```

Comment:
Alternatively, we should avoid using variables so we can document the hooks correctly?